tls/s2n_resume.c

```diff
@@ -563,6 +561,67 @@ int s2n_decrypt_session_ticket(struct s2n_connection *conn)
     return 0;
 }
 
+int s2n_encrypt_session_cache(struct s2n_connection *conn, struct s2n_stuffer *to)
+{
+    return s2n_encrypt_session_ticket(conn, to);
+}
+
+
+int s2n_decrypt_session_cache(struct s2n_connection *conn, struct s2n_stuffer *from)
+{
+    struct s2n_ticket_key *key;
+    struct s2n_session_key aes_ticket_key;
+    struct s2n_blob aes_key_blob;
+
+    uint8_t key_name[S2N_TICKET_KEY_NAME_LEN];
+
+    uint8_t iv_data[S2N_TLS_GCM_IV_LEN] = { 0 };
+    struct s2n_blob iv = { .data = iv_data, .size = sizeof(iv_data) };
+
+    uint8_t aad_data[S2N_TICKET_AAD_LEN] = { 0 };
+    struct s2n_blob aad_blob = { .data = aad_data, .size = sizeof(aad_data) };
+    struct s2n_stuffer aad;
+
+    uint8_t s_data[S2N_STATE_SIZE_IN_BYTES] = { 0 };
+    struct s2n_blob state_blob = { .data = s_data, .size = sizeof(s_data) };
+    struct s2n_stuffer state;
+
+    uint8_t en_data[S2N_STATE_SIZE_IN_BYTES + S2N_TLS_GCM_TAG_LEN];
+    struct s2n_blob en_blob = { .data = en_data, .size = sizeof(en_data) };
+
+    GUARD(s2n_stuffer_read_bytes(from, key_name, S2N_TICKET_KEY_NAME_LEN));
+
+    key = s2n_find_ticket_key(conn->config, key_name);
+
+    /* Key has expired; do full handshake with NST */
```

Comment:
NST?

Reply:
New Session Ticket (NST). Update in comment